### ISSUE TITLE
PyYaml not installed by default causes incorrect reporting to Coveralls

### ISSUE THREAD

**Issue by devdazed** [coveralls-clients/coveralls-python#99]
ref: https://github.com/lemurheavy/coveralls-public/issues/657

Currently the `PyYAML` package is declared as an extra and not installed by default.  In a new project that does not use  any YAML dependency, the `.coveralls.yml` file load silently fails and default to an empty dictionary.  This causes reporting issues like `travis-pro` to report as `travis-ci`.  Since Travis-CI and Travis-Pro have job-id conflicts between each other, this will lead to incorrect commit messages and incorrect committer information reported in Coveralls.

With the following `.coveralls.yml`

``` yml
service_name: travis-pro
parallel: true
```

I was seeing things like this in Coveralls:
![screen shot 2015-11-06 at 10 43 43 am](https://cloud.githubusercontent.com/assets/413041/11001075/49e7e4ba-8473-11e5-98d2-d65a33ee1306.png)

After some digging I found that the config is loaded like this:

``` python
    def load_config(self):
        try:
            import yaml
            return yaml.safe_load(open(os.path.join(os.getcwd(), self.config_filename)))
        except ImportError:
            return {}
        except (OSError, IOError):
            log.debug('Missing %s file. Using only env variables.', self.config_filename)
```

The `ImportError` silently fails, and the user is not aware that there is any issue.

**Comment by coagulant** [comment]
* `.coveralls.yml` is supported, but not documented, this clearly needs to be fixed
* if `.coveralls.yml` is present, but coveralls[yaml] is not installed, coveralls need to emit a warning, this is also a bug.

Thnx for detailed explanation, PR is welcome

**Comment by maxkoryukov** [comment]
I got the same bug. I have a solution: #104

**Comment by TheKevJames** [comment]
1. I just created documentation for the `.coveralls.yml` file. It is [here](http://coveralls-python.readthedocs.io/en/latest/usage/configuration.html).
2. Fixed in #104
